call_host.sh

```diff
@@ -8,15 +9,70 @@ if [ -f "$CALL_HOST_CONFIG" ]; then
 	source "$CALL_HOST_CONFIG"
 fi
 
+# zsh / bash compatibility helpers
+is_zsh(){
+	# detect the current shell process name (portable ps usage)
+	if command -v ps >/dev/null 2>&1; then
+		# get last path component if ps returns full path
+		p="$(ps -p $$ -o comm= 2>/dev/null | awk -F/ '{print $NF}')"
+		case "$p" in
+			zsh) return 0 ;;
+		esac
+	fi
+
+	# fallback: check common names for $0 or $ZSH_NAME (login shells may have a leading dash)
+	case "$(basename -- "${ZSH_NAME:-$0}" 2>/dev/null)" in
+		zsh|-zsh) return 0 ;;
+	esac
+
+	return 1
+}
+
+if is_zsh; then
+	# export a function to the environment for child shells (zsh)
+	export_func(){
+		typeset -fx "$1" 2>/dev/null || true
+	}
+	# declare an associative array (zsh) - create a named array using eval so dynamic name works
+	declare_assoc(){
+		eval "typeset -A $1"
```

Comment:
check if eval necessary here

Reply:
If I just enter `/bin/zsh`, the following works as the similar test did above for bash:
```zsh
TEST1=TEST2
typeset -A $TEST1
TEST2[foo]=bar
echo "${TEST2[@]}"
```
output:
```
bar
```